lib/std/http/Client.zig

```diff
@@ -1434,6 +1434,7 @@ pub fn connectTunnel(
     proxy: *Proxy,
     tunnel_host: []const u8,
     tunnel_port: u16,
+    tunnel_protocol: Connection.Protocol,
 ) !*Connection {
     if (!proxy.supports_connect) return error.TunnelNotSupported;
 
```

Comment:
@tw4452852 I'm thinking we should also fix this too. under this line.
how do you think?

```diff
 if (client.connection_pool.findConnection(.{
         .host = tunnel_host,
         .port = tunnel_port,
-        .protocol = proxy.protocol,
+        .protocol = tunnel_protocol,
     })) |node|
         return node;
```